### PR TITLE
NPM Analyzer NullPointerException fix

### PR DESCRIPTION
**Pull request by andreaslarfors**
Correcting possible nullpointerException in NPM Analyzer where package information is fetched OK but does not contain any version entries.

Signed-off-by: Andreas Lärfors <andreas@larfors.com>

### CODE REVIEW

File: analyzer/src/main/kotlin/managers/Npm.kt [hunk 1]
```diff
@@ -252,26 +252,31 @@ open class Npm(
...
-
-                                versionInfo["dist"]?.let { dist ->
-                                    downloadUrl = dist["tarball"].textValueOrEmpty().let { tarballUrl ->
-                                        if (tarballUrl.startsWith("http://registry.npmjs.org/")) {
-                                            // Work around the issue described at
-                                            // https://npm.community/t/some-packages-have-dist-tarball-as-http-and-not-https/285/19.
-                                            "https://" + tarballUrl.removePrefix("http://")
-                                        } else {
-                                            tarballUrl
+                            if (packageInfo["versions"] != null) {
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/Npm.kt:255]: When you say in the commit message "does not contain any version entries" I would have expected that `packageInfo["versions"]` is present but empty. However, here you're checking whether `packageInfo["versions"]` is not present at all.

Anyway, the case that `packageInfo["versions"]` is present but empty is already covered by `?.` after `[version]`, so we really only need to check for `packageInfo["versions"]` not being present at all, which we can do without increasing the nesting level by doing

    packageInfo["versions"]?.get(version)?.let { versionInfo ->

Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/Npm.kt:255]: Also you commit message needs for rework:

- Please do not exceed the title line beyond about 75 chars,
- prefix the title with the "scope" of your changes, which for single-file changes is just the file name (without extension),
- write "nullpointerException" as "NullPointerException".

So how about this:

> Npm: Avoid a NullPointerException if the "versions" node is not present

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: analyzer/src/main/kotlin/managers/Npm.kt (hunk 1 above)
Comment (andreaslarfors) [reply analyzer/src/main/kotlin/managers/Npm.kt]: Thanks @sschuberth, agreed on all points. Commit messages amended and code refactored accordingly.

Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/Npm.kt]: Unfortunately, now the Git history is messed up as it contains 4 (!) commits. Please squash them all into one commit (you should have just amended your existing commit and force-pushed to address review comments).

**Comment by sschuberth** [comment]
@andreaslarfors is there still interested from your side to work on this? What's missing for me simply is to squash all commits into one, and maybe rebase one latest *master*.

**Comment by sschuberth** [comment]
Closing this due to a lack of feedback in favor of https://github.com/oss-review-toolkit/ort/pull/3466.

**Pull request closed without merging**